CONTRIBUTING.md

```diff
@@ -1,84 +1,90 @@
+The version number of the Python package, defined in
+`src/pg/crankshaft/setup.py` will be overridden when
+the package is released and always match the extension version number,
+but for development it shall be kept as '0.0.0'.
 
-...
+Run the tests with `make test`.
 
-### SQL Extension
+To use the python extension for custom tests, activate the virtual
+environment with:
 
-* Generate a **new subfolder version** for `sql` and `test` folders to define
-  the new functions and tests
-  - Use symlinks to avoid file duplication between versions that don't update them
-  - Add new files or modify copies of the old files to add new functions or
-    modify existing functions (remember to rename a function if the signature
-    changes)
-  - Add or modify the corresponding documentation files in the `doc` folder.
-    Since we expect to have highly technical functions here, an extense
-    background explanation would be of great help to users of this extension.
-  - Create tests for the new functions/behaviour
-
-* Generate the **upgrade and downgrade files** for the extension
-
-* Update the control file and the Makefile to generate the complete SQL
-  file for the new created version. After running `make` a new
-  file `crankshaft--X.Y.Z.sql` will be created for the current version.
-  Additional files for migrating to/from the previous version A.B.Z should be
-  created:
-  - `crankshaft--X.Y.Z--A.B.C.sql`
-  - `crankshaft--A.B.C--X.Y.Z.sql`
-  All these new files must be added to git and pushed.
+```
+source envs/dev/bin/activate
+```
 
-* Update the public docs! ;-)
+Update extension in a working database with:
 
-## Conventions
+* `ALTER EXTENSION crankshaft VERSION TO 'current';`
+  `ALTER EXTENSION crankshaft VERSION TO 'dev';`
 
-# SQL
+Note: we keep the current development version install as 'dev' always;
+we update through the 'current' alias to allow changing the extension
+contents but not the version identifier. This will fail if the
+changes involve incompatible function changes such as a different
+return type; in that case the offending function (or the whole extension)
+should be dropped manually before the update.
 
-Use snake case (i.e. `snake_case` and not `CamelCase`) for all
-functions. Prefix functions intended for public use with `cdb_`
-and private functions (to be used only internally inside
-the extension)  with `_cdb_`.
+If the extension has not previously been installed in a database,
+it can be installed directly with:
 
-# Python
+* `CREATE EXTENSION crankshaft WITH VERSION 'dev';`
 
-...
+Note: the development extension uses the development python virtual
+environment automatically.
 
-## Testing
+Before proceeding to the release process peer code reviewing of the code is
+a must.
 
-Running just the Python tests:
+Once the feature or bugfix is completed, all the tests are passing
+and the code has been accepted by peer reviewing,
+the topic branch can be merged back into the `develop` branch and a
```

Comment:
Actually the order facilitated by github is PR first, then accept-merge the PR "a PR shall be created, reviewed by a peer and then merged to `develop` when CI tests pass"